src/simple_minreq.rs

```diff
@@ -0,0 +1,271 @@
+//! This module implements the [`crate::client::Transport`] trait using [minreq] as the underlying
+//! HTTP transport.
+//!
+//! [minreq]: <https://github.com/neonmoe/minreq>
+//!
+
+#[cfg(not(fuzzing))]
+use std::time::Duration;
+use std::{error, fmt};
+
+use crate::client::Transport;
+use crate::{Request, Response};
+
+/// Global mutex used by the fuzzing harness to inject data into the read
+/// end of the TCP stream.
+#[cfg(fuzzing)]
+pub static FUZZ_TCP_SOCK: Mutex<Option<io::Cursor<Vec<u8>>>> = Mutex::new(None);
+
+#[cfg(fuzzing)]
+#[derive(Clone, Debug)]
+struct TcpStream;
+
+#[cfg(fuzzing)]
+mod impls {
+    use super::*;
+    impl Read for TcpStream {
+        fn read(&mut self, buf: &mut [u8]) -> io::Result<usize> {
+            match *FUZZ_TCP_SOCK.lock().unwrap() {
+                Some(ref mut cursor) => io::Read::read(cursor, buf),
+                None => Ok(0),
+            }
+        }
+    }
+    impl Write for TcpStream {
+        fn write(&mut self, buf: &[u8]) -> io::Result<usize> {
+            io::sink().write(buf)
+        }
+        fn flush(&mut self) -> io::Result<()> {
+            Ok(())
+        }
+    }
+
+    impl TcpStream {
+        pub fn connect_timeout(_: &SocketAddr, _: Duration) -> io::Result<Self> {
+            Ok(TcpStream)
+        }
+        pub fn set_read_timeout(&self, _: Option<Duration>) -> io::Result<()> {
+            Ok(())
+        }
+        pub fn set_write_timeout(&self, _: Option<Duration>) -> io::Result<()> {
+            Ok(())
+        }
+    }
+}
+
+/// The default TCP port to use for connections.
+/// Set to 8332, the default RPC port for bitcoind.
+pub const DEFAULT_PORT: u16 = 8332;
+
+/// Simple HTTP transport that implements the necessary subset of HTTP for
+/// running a bitcoind RPC client.
+#[derive(Clone, Debug)]
+pub struct SimpleHttpTransport {
```

Comment:
In d20ba2eafcaa3cd5245f5a548cbcbb6d14c13432:

This struct should not be called `SimpleHttpTransport`.